tests/integration/src/test/resources/set-variable-mime-type-config.xml

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns="http://www.mulesoft.org/schema/mule/core"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:vm="http://www.mulesoft.org/schema/mule/vm"
+      xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+       http://www.mulesoft.org/schema/mule/vm http://www.mulesoft.org/schema/mule/vm/current/mule-vm.xsd">
+
+    <flow name="mimeTypeOnly">
+        <vm:inbound-endpoint path="testInput1" exchange-pattern="request-response"/>
+
+        <set-variable variableName="testVariable" value="#['test']" encoding="UTF-16" mimeType="application/json"/>
+
+        <component class="org.mule.transformers.simple.SetVariableMimeTypeTestCase$FlowVariableDataTypeAccessor"/>
+    </flow>
+
+    <flow name="contentTypeOnly">
+        <vm:inbound-endpoint path="testInput2" exchange-pattern="request-response"/>
+
+        <set-variable variableName="testVariable" value="#['test']" encoding="UTF-16"/>
+
+        <set-property propertyName="Content-Type" value="application/json;charset=UTF-16BE"/>
```

Comment:
I don't think this is reproducing the payload test case, I think the ideal thing to do would be to send the Content type on the client request instead of setting it on the app

Reply:
Removed this property from the app and added setOutboundProperty to request instead.

Reply:
Notice that you're adding the property to the response not to the request, after the flow has run. Maybe there's some way to add the content-type when sending the request

Reply:
Used props map instead to have a more relevant test case as suggested.